lightningd/gossip_control.c

```diff
@@ -512,6 +512,23 @@ static void json_getnodes_reply(struct subd *gossip UNUSED, const u8 *reply,
 			json_add_address(response, NULL, &nodes[i]->addresses[j]);
 		}
 		json_array_end(response);
+		if (nodes[i]->ad) {
+			json_object_start(response, "liquidity_ad");
+			json_add_num(response,
+				     "lease_fee_basis",
+				     nodes[i]->ad->lease_basis);
+			json_add_amount_sat_only(response,
+						 "lease_fee_base_msat",
+						 nodes[i]->ad->lease_base_sat);
+			json_add_num(response,
+				     "channel_fee_basis_max",
+				     nodes[i]->ad->channel_fee_basis);
+			json_add_amount_msat_only(response,
+						  "channel_fee_base_max_msat",
+						  nodes[i]->ad->channel_base_msat);
+
```

Comment:
Needs weight.

Reply:
I don't think including a max weight in the offer card is a good idea -- a smart node is *always* going to have their peer cover their weight of opening the tx, and that's going to change depending on what their current UTXO set is + the offer requested by the peer.

Consider the case where your advertised weight is 100vbytes, but it's going to cost you 120vbytes to fund a channel for the amount a peer has requested. You're not going to cover that extra 20vbytes. The logical thing would be to update the `lease_fee_base` by the difference to that peer to cover the additional cost.

Pre-committing to the weight means that the other numbers won't be reliable, which will make it harder to do an apples to apples comparison of offers across the board. (which, tbf aren't going to be apples to apples until you get their weights back, but you can't know what that'll be until you ask....). For this reason, we should leave the weight out of the offer advertisement, and allow it to be 'discoverable' at open time -- then it's just a matter of setting sane limits on the opener side for an acceptable weight you're willing to pay for.

otoh it *is* a lot easier to evaluate offers if you know the cost of them all upfront ........

Ok fine you've talked me into it. I'll add a "max weight charge" to the rate card, with the idea that a node should never ask for *more* than this. Good talk!